### ISSUE TITLE
cmst: fails to compile with meta-qt6

### ISSUE THREAD

**Issue by ansar-rasool** [andrew-bibb/cmst#277]
I am trying to build cmst with qt6 and yocto 4.0 (kirkstone). But cmst fails to compile and throws following error. Is cmst supported with qt6? or are there any plans in future to support with latest qt version?

```
DEBUG: Executing shell function do_compile
NOTE: make -j 48 MAKEFLAGS=-j 48 OE_QMAKE_CC=aarch64-flex-linux-gcc  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot OE_QMAKE_CXX=aarch64-flex-linux-g++  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot OE_QMAKE_CFLAGS= -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  OE_QMAKE_CXXFLAGS= -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden OE_QMAKE_LINK=aarch64-flex-linux-g++  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot OE_QMAKE_LDFLAGS=-Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  OE_QMAKE_AR=aarch64-flex-linux-gcc-ar OE_QMAKE_STRIP=echo OE_QMAKE_INCDIR_QT=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot//usr/include
cd apps/cmstapp/ && ( test -e Makefile || /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native/usr/bin/qmake -o Makefile /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/git/apps/cmstapp/cmstapp.pro DISTRO=debian ) && make -f Makefile 
cd apps/rootapp/ && ( test -e Makefile || /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native/usr/bin/qmake -o Makefile /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/git/apps/rootapp/rootapp.pro DISTRO=debian ) && make -f Makefile 
make[1]: Entering directory '/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/git/apps/rootapp'
make[1]: Nothing to be done for 'first'.
make[1]: Leaving directory '/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/git/apps/rootapp'
make[1]: Entering directory '/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/git/apps/cmstapp'
aarch64-flex-linux-g++  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot -c -pipe -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -O2 -Wall -Wextra -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I../../../recipe-sysroot/usr/include/QtWidgets -I../../../recipe-sysroot/usr/include/QtGui -I../../../recipe-sysroot/usr/include/QtDBus -I../../../recipe-sysroot/usr/include/QtNetwork -I../../../recipe-sysroot/usr/include/QtCore -Imoc_files -I. -I../../../recipe-sysroot/usr/lib/mkspecs/linux-oe-g++ -o object_files/main.o code/main.cpp
aarch64-flex-linux-g++  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot -c -pipe -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -O2 -Wall -Wextra -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I../../../recipe-sysroot/usr/include/QtWidgets -I../../../recipe-sysroot/usr/include/QtGui -I../../../recipe-sysroot/usr/include/QtDBus -I../../../recipe-sysroot/usr/include/QtNetwork -I../../../recipe-sysroot/usr/include/QtCore -Imoc_files -I. -I../../../recipe-sysroot/usr/lib/mkspecs/linux-oe-g++ -o object_files/controlbox.o code/control_box/controlbox.cpp
aarch64-flex-linux-g++  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot -c -pipe -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -O2 -Wall -Wextra -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I../../../recipe-sysroot/usr/include/QtWidgets -I../../../recipe-sysroot/usr/include/QtGui -I../../../recipe-sysroot/usr/include/QtDBus -I../../../recipe-sysroot/usr/include/QtNetwork -I../../../recipe-sysroot/usr/include/QtCore -Imoc_files -I. -I../../../recipe-sysroot/usr/lib/mkspecs/linux-oe-g++ -o object_files/agent.o code/agent/agent.cpp
aarch64-flex-linux-g++  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot -c -pipe -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -O2 -Wall -Wextra -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I../../../recipe-sysroot/usr/include/QtWidgets -I../../../recipe-sysroot/usr/include/QtGui -I../../../recipe-sysroot/usr/include/QtDBus -I../../../recipe-sysroot/usr/include/QtNetwork -I../../../recipe-sysroot/usr/include/QtCore -Imoc_files -I. -I../../../recipe-sysroot/usr/lib/mkspecs/linux-oe-g++ -o object_files/counter.o code/counter/counter.cpp
aarch64-flex-linux-g++  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot -c -pipe -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -O2 -Wall -Wextra -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I../../../recipe-sysroot/usr/include/QtWidgets -I../../../recipe-sysroot/usr/include/QtGui -I../../../recipe-sysroot/usr/include/QtDBus -I../../../recipe-sysroot/usr/include/QtNetwork -I../../../recipe-sysroot/usr/include/QtCore -Imoc_files -I. -I../../../recipe-sysroot/usr/lib/mkspecs/linux-oe-g++ -o object_files/vpnagent.o code/vpn_agent/vpnagent.cpp
aarch64-flex-linux-g++  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot -c -pipe -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -O2 -Wall -Wextra -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I../../../recipe-sysroot/usr/include/QtWidgets -I../../../recipe-sysroot/usr/include/QtGui -I../../../recipe-sysroot/usr/include/QtDBus -I../../../recipe-sysroot/usr/include/QtNetwork -I../../../recipe-sysroot/usr/include/QtCore -Imoc_files -I. -I../../../recipe-sysroot/usr/lib/mkspecs/linux-oe-g++ -o object_files/vpnagent_adaptor.o code/vpn_agent/vpnagent_adaptor.cpp
aarch64-flex-linux-g++  -march=armv8-a+crc --sysroot=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot -c -pipe -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0=/usr/src/debug/cmst/git-r0                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot=                      -fdebug-prefix-map=/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -O2 -Wall -Wextra -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_DBUS_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -I../../../recipe-sysroot/usr/include/QtWidgets -I../../../recipe-sysroot/usr/include/QtGui -I../../../recipe-sysroot/usr/include/QtDBus -I../../../recipe-sysroot/usr/include/QtNetwork -I../../../recipe-sysroot/usr/include/QtCore -Imoc_files -I. -I../../../recipe-sysroot/usr/lib/mkspecs/linux-oe-g++ -o object_files/qrc_cmst.o qrc_cmst.cpp
/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native/usr/bin/lrelease ../../translations/cmst_pt_BR.ts -qm .qm/cmst_pt_BR.qm
make[1]: /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot-native/usr/bin/lrelease: Command not found
make[1]: *** [Makefile:515: .qm/cmst_pt_BR.qm] Error 127
make[1]: *** Waiting for unfinished jobs....
code/control_box/controlbox.cpp:60:11: fatal error: QDesktopWidget: No such file or directory
   60 | # include <QDesktopWidget>
      |           ^~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:1282: object_files/controlbox.o] Error 1
In file included from code/vpn_agent/vpnagent_adaptor.cpp:11:
code/vpn_agent/vpnagent_adaptor.h:22:7: error: using typedef-name 'using QStringList = class QList<QString>' after 'class'
   22 | class QStringList;
      |       ^~~~~~~~~~~
In file included from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qtypeinfo.h:42,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qglobal.h:1431,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qnamespace.h:44,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qobjectdefs.h:48,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qobject.h:46,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/QObject:1,
                 from code/vpn_agent/vpnagent_adaptor.h:15,
                 from code/vpn_agent/vpnagent_adaptor.cpp:11:
/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qcontainerfwd.h:64:7: note: 'using QStringList = class QList<QString>' has a previous declaration here
   64 | using QStringList = QList<QString>;
      |       ^~~~~~~~~~~
In file included from code/vpn_agent/vpnagent.cpp:42:
code/vpn_agent/./vpnagent_adaptor.h:22:7: error: using typedef-name 'using QStringList = class QList<QString>' after 'class'
   22 | class QStringList;
      |       ^~~~~~~~~~~
In file included from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qtypeinfo.h:42,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qglobal.h:1431,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qalgorithms.h:43,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qdebug.h:44,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/QDebug:1,
                 from code/vpn_agent/vpnagent.cpp:30:
/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qcontainerfwd.h:64:7: note: 'using QStringList = class QList<QString>' has a previous declaration here
   64 | using QStringList = QList<QString>;
      |       ^~~~~~~~~~~
make[1]: *** [Makefile:1696: object_files/vpnagent_adaptor.o] Error 1
code/main.cpp: In function 'int main(int, char**)':
code/main.cpp:58:39: warning: 'Qt::AA_UseHighDpiPixmaps' is deprecated: High-DPI pixmaps are always enabled. This attribute no longer has any effect. [-Wdeprecated-declarations]
   58 |    QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);
      |                                       ^~~~~~~~~~~~~~~~~~~~
In file included from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qbytearray.h:45,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qstringview.h:58,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qchar.h:666,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qstring.h:50,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qhashfunctions.h:44,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qhash.h:45,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qdebug.h:45,
                 from /data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/QDebug:1,
                 from code/main.cpp:29:
/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/recipe-sysroot/usr/include/QtCore/qnamespace.h:464:9: note: declared here
  464 |         AA_UseHighDpiPixmaps Q_DECL_ENUMERATOR_DEPRECATED_X(
      |         ^~~~~~~~~~~~~~~~~~~~
code/main.cpp:173:18: warning: 'static QString QLibraryInfo::location(QLibraryInfo::LibraryLocation)' is deprecated: Use path() [-Wdeprecated-declarations]
  173 |    QLibraryInfo::location(QLibraryInfo::TranslationsPath));
      |                  ^~~~~~~~
In file included from ../../../recipe-sysroot/usr/include/QtCore/QLibraryInfo:1,
                 from code/main.cpp:38:
../../../recipe-sysroot/usr/include/QtCore/qlibraryinfo.h:83:20: note: declared here
   83 |     static QString location(LibraryLocation location)
      |                    ^~~~~~~~
code/main.cpp:173:26: warning: 'static QString QLibraryInfo::location(QLibraryInfo::LibraryLocation)' is deprecated: Use path() [-Wdeprecated-declarations]
  173 |    QLibraryInfo::location(QLibraryInfo::TranslationsPath));
      |    ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../../recipe-sysroot/usr/include/QtCore/QLibraryInfo:1,
                 from code/main.cpp:38:
../../../recipe-sysroot/usr/include/QtCore/qlibraryinfo.h:83:20: note: declared here
   83 |     static QString location(LibraryLocation location)
      |                    ^~~~~~~~
code/main.cpp:172:21: warning: ignoring return value of 'bool QTranslator::load(const QString&, const QString&, const QString&, const QString&)', declared with attribute 'nodiscard' [-Wunused-result]
  172 |    qtTranslator.load("qt_" + QLocale::system().name(),
      |    ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  173 |    QLibraryInfo::location(QLibraryInfo::TranslationsPath));
      |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../../recipe-sysroot/usr/include/QtCore/QTranslator:1,
                 from code/main.cpp:37:
../../../recipe-sysroot/usr/include/QtCore/qtranslator.h:69:24: note: declared here
   69 |     [[nodiscard]] bool load(const QString & filename,
      |                        ^~~~
make[1]: *** [Makefile:1663: object_files/vpnagent.o] Error 1
make[1]: Leaving directory '/data/ansar/ginkgo/build-qemu_qt/tmp/work/armv8a-flex-linux/cmst/git-r0/git/apps/cmstapp'
make: *** [Makefile:48: sub-apps-cmstapp-make_first] Error 2
ERROR: oe_runmake failed
WARNING: exit code 1 from a shell command.
```

**Comment by andrew-bibb** [comment]
I have not tried QT-6 yet.  For the last year at least the 5 series has been marking a selection of functions depreciated in anticipation of QT-6 and as I ran across each I updated it to the replacement function.  I hope that will make the transition from 5 to 6 easier.

This will be my first priority when I start back on programming, likely in September or October.  I live in an area with short summers and don't tend to work at my computer in the warmer months.
